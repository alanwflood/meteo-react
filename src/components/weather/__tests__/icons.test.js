import React from "react";
import Icon from "../icon";
import { render } from "@testing-library/react";

describe("Weather icon component", () => {
  it("Renders the Clear Day icon", () => {
    const { container } = render(<Icon icon="01d" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Clear Night icon", () => {
    const { container } = render(<Icon icon="01n" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Day icon", () => {
    const { container } = render(<Icon icon="02d" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Night icon", () => {
    const { container } = render(<Icon icon="02n" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Cloudy Day icon", () => {
    const { container } = render(<Icon icon="03d" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Cloudy Night icon", () => {
    const { container } = render(<Icon icon="03n" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Cloudy icon", () => {
    const { container } = render(<Icon icon="04" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Rain icon", () => {
    const { container } = render(<Icon icon="09" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Heavy Rain icon", () => {
    const { container } = render(<Icon icon="10" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Storm icon", () => {
    const { container } = render(<Icon icon="11" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Snow icon", () => {
    const { container } = render(<Icon icon="13" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Hail icon", () => {
    const { container } = render(<Icon icon="50" />);
    expect(container).toMatchSnapshot();
  });
  it("Renders the Day icon by default", () => {
    const { container } = render(<Icon />);
    expect(container).toMatchSnapshot();
  });
});
