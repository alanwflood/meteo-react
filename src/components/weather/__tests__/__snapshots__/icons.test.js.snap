// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather icon component Renders the Clear Day icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Clear Night icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Cloudy Day icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Cloudy Night icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Cloudy icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Day icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Day icon by default 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Hail icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Heavy Rain icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Night icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Rain icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Snow icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;

exports[`Weather icon component Renders the Storm icon 1`] = `
<div>
  <div
    class="icon weather-icon"
  >
    <test-file-stub />
  </div>
</div>
`;
